core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -133,6 +136,67 @@ public void testDynamicHttpRequestInterceptorLoading() {
     assertThat(((TestHttpRequestInterceptor) interceptor).properties).isEqualTo(properties);
   }
 
+  @Test
+  public void testSocketAndConnectionTimeoutSet() {
+    long connectionTimeoutMs = 10L;
+    int socketTimeoutMs = 10;
+    Map<String, String> properties =
+        ImmutableMap.of(
+            HTTPClient.REST_CONNECTION_TIMEOUT_MS, String.valueOf(connectionTimeoutMs),
+            HTTPClient.REST_SOCKET_TIMEOUT_MS, String.valueOf(socketTimeoutMs));
+
+    ConnectionConfig connectionConfig = HTTPClient.configureConnectionConfig(properties);
+    assertThat(connectionConfig).isNotNull();
+    assertThat(connectionConfig.getConnectTimeout().getDuration()).isEqualTo(connectionTimeoutMs);
+    assertThat(connectionConfig.getSocketTimeout().getDuration()).isEqualTo(socketTimeoutMs);
+  }
+
+  @Test
+  public void testSocketTimeout() throws IOException {
+    long socketTimeoutMs = 2000L;
+    Map<String, String> properties =
+        ImmutableMap.of(HTTPClient.REST_SOCKET_TIMEOUT_MS, String.valueOf(socketTimeoutMs));
+    String path = "socket/timeout/path";
+
+    try (HTTPClient client = HTTPClient.builder(properties).uri(URI).build()) {
+      HttpRequest mockRequest =
+          request()
+              .withPath("/" + path)
+              .withMethod(HttpMethod.HEAD.name().toUpperCase(Locale.ROOT));
+      // Setting a response delay of 5 seconds to simulate hitting the configured socket timeout of
+      // 2 seconds
+      HttpResponse mockResponse =
+          response()
+              .withStatusCode(200)
+              .withBody("Delayed response")
+              .withDelay(TimeUnit.MILLISECONDS, 5000);
+      mockServer.when(mockRequest).respond(mockResponse);
+
+      Assertions.assertThatThrownBy(() -> client.head(path, ImmutableMap.of(), (unused) -> {}))
+          .cause()
+          .isInstanceOf(SocketTimeoutException.class);
+    }
+  }
+
+  @Test
+  public void testInvalidConnectionTimeout() {
+    testInvalidTimeouts(HTTPClient.REST_CONNECTION_TIMEOUT_MS);
+  }
+
+  @Test
+  public void testInvalidSocketTimeout() {
+    testInvalidTimeouts(HTTPClient.REST_SOCKET_TIMEOUT_MS);
+  }
+
+  private void testInvalidTimeouts(String timeoutMsType) {
```

Comment:
```suggestion
  private void checkInvalidTimeout(String timeoutMsType) {
```

Reply:
you can then remove `testInvalidConnectionTimeout` / `testInvalidSocketTimeout`

Reply:
I'm so sorry for the repeated reviews. Fixed.